analysis/src/xml.rs

```diff
@@ -0,0 +1,715 @@
+use roxmltree::StringStorage;
+use std::borrow::Cow;
+
+/// A node with its `'input` lifetime set to `'static`.
+type Node<'a> = roxmltree::Node<'a, 'static>;
+/// String type used for XML attribute and text values.
+///
+/// A `&'static str` is not used directly because sometimes string allocation is
+/// needed, for example when replacing `&quot;` with normal quotes.
+type XmlStr = Cow<'static, str>;
+
+/// Retrieves the value of the `node`'s attribute named `name`.
+fn attribute(node: Node, name: &str) -> Option<XmlStr> {
+    node.attribute_node(name)
+        .map(|attr| match attr.value_storage().clone() {
+            StringStorage::Borrowed(s) => Cow::Borrowed(s),
+            StringStorage::Owned(s) => Cow::Owned((*s).into()),
+        })
+}
+
+/// Retrieves the text inside the next child element of `node` named `name`.
+fn child_text(node: Node, name: &str) -> Option<XmlStr> {
+    let child = node.children().find(|node| node.has_tag_name(name));
```

Comment:
Do you think we need more `StringStorage` -> `Cow` conversions, and should we factor it out into a function?

Reply:
There probably won't be any more conversions. We could factor it out, but I don't think it's really worth it currently.